src/main/java/io/cdap/plugin/kinesis/streaming/ReferenceStreamingSource.java

```diff
@@ -58,4 +64,23 @@ public void run(DatasetContext datasetContext) {
     });
   }
 
+  /**
+   * Record field-level lineage for streaming source plugins (ReadOperation). This method should be called from
+   * prepareRun of any streaming source plugin.
+   * @param context StreamingSourceContext from prepareRun
+   * @param outputName name of output dataset
+   * @param tableSchema schema of fields. Also used to determine list of field names. Schema and schema.getFields() must
+   * not be null.
+   * @param operationName name of the operation
+   * @param description operation description; complete sentences preferred
+   */
+  protected void recordLineage(StreamingSourceContext context, String outputName, Schema tableSchema,
+                               String operationName, String description) {
+    LineageRecorder lineageRecorder = new LineageRecorder(context, outputName);
+    lineageRecorder.createExternalDataset(tableSchema);
```

Comment:
streaming source does not need this, instead it should call https://github.com/cdapio/hydrator-plugins/blob/develop/spark-plugins/src/main/java/io/cdap/plugin/spark/FileStreamingSource.java#L74